File: android/app/src/main/java/com/baatcheet/app/domain/usecase/auth/LogoutUseCase.kt
package com.baatcheet.app.domain.usecase.auth

import com.baatcheet.app.domain.repository.AuthRepository
import javax.inject.Inject

/**
 * Logout Use Case
 * 
 * Handles the business logic for user logout.
 */
class LogoutUseCase @Inject constructor(
    private val authRepository: AuthRepository
) {
    /**
     * Execute logout
     */
    suspend operator fun invoke(): Boolean {
        return authRepository.logout()
    }
}

File: ios/iOSBaseProject/Domain/UseCases/Auth/LogoutUseCase.swift
//
//  LogoutUseCase.swift
//  BaatCheet
//
//  Domain layer - Logout Use Case
//

import Foundation

/// Use case for logging out the user.
protocol LogoutUseCaseProtocol {
    func execute() async
}

final class LogoutUseCase: LogoutUseCaseProtocol {
    private let repository: AuthRepositoryProtocol
    
    init(repository: AuthRepositoryProtocol) {
        self.repository = repository
    }
    
    func execute() async {
        await repository.signOut()
    }
}
